pkcs1/rsassa_pkcs1_v15.py

```diff
@@ -49,6 +49,8 @@ def verify(public_key, message, signature, hash_class=hashlib.sha1):
     try:
         em_prime = emsa_pkcs1_v15.encode(message, public_key.byte_size,
                 hash_class=hash_class)
+        em_prime_imp = emsa_pkcs1_v15.encode(message, public_key.byte_size,
+                hash_class=hash_class, explicit_null_param=False)
     except ValueError:
         raise exceptions.RSAModulusTooShort
-    return primitives.constant_time_cmp(em, em_prime)
+    return primitives.constant_time_cmp(em, em_prime) or primitives.constant_time_cmp(em, em_prime_imp)
```

Comment:
Is this still constant time?

In an `or` statement, Python only evaluates the second condition if the first condition evaluates to `True`. See the [Python specification for Boolean operations.](https://docs.python.org/3/reference/expressions.html#boolean-operations).

Reply:
You are right. But don't expect this lib to give you real constant time even with this change. It's a toy implementation, if you need a really hardened implementation of RSA, there are a lot around written in C or ASM.